test/iterator/basic_iterator.cpp

```diff
@@ -598,6 +622,34 @@ void test_proxy_array() {
 	CHECK(a[3] == 1);
 }
 
+void test_output() {
+	std::cout << "test_output:\n";
+
+	auto vec = std::vector<int>{0,0,0};
+	auto wi = vec.begin();
+
+	using I = ::output_iterator<decltype(wi)>;
+	static_assert(stl2::models::WeaklyIncrementable<I>);
+	static_assert(!stl2::models::Incrementable<I>);
+	static_assert(!stl2::models::Decrementable<I>);
+	static_assert(!stl2::models::Readable<I>);
+	static_assert(stl2::models::Writable<I, int>);
+	static_assert(stl2::models::DefaultConstructible<I>);
+	static_assert(stl2::models::Copyable<I>);
+	static_assert(stl2::models::Semiregular<I>);
+	static_assert(stl2::models::Iterator<I>);
+	static_assert(!stl2::models::InputIterator<I>);
+	static_assert(stl2::models::OutputIterator<I, int>);
+
+	I i{wi};
+	*i++ = 1;
+	*i++ = 2;
+	*i++ = 3;
+	CHECK(vec[0] == 1);
+	CHECK(vec[1] == 2);
+	CHECK(vec[2] == 3);
```

Comment:
Actually, separating increment and dereference in this way results in undefined behavior for output iterators that wrap other output iterators if the post-increment operator doesn't return a self reference.  In this case, the test is exercising an output iterator wrapping a random access iterator, so it isn't a problem.  If the wrapped iterator was an actual output iterator, then the post-increment operation would have incremented the wrapped iterator held by <tt>i</tt>, the dereference operation would have dereferenced the wrapped iterator in the copy of <tt>i</tt> returned by the post-increment operation, and any subsequent increment operation on <tt>i</tt> would be undefined behavior.  With the current design, the best way I see to avoid that undefined behavior is to implement <tt>post_increment</tt> to return a proxy that implements the dereference and assignment operators.  I suppose that is ok, but since it is non-trivial to do, having a solution that allows a cursor to opt-in to having post-increment return a self reference seems desirable.